card-set/src/card-component.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import './Card.css';
+
+const UserCard = (props) => {
+  const [avatar, setAvatar] = React.useState('https://media.tenor.com/images/e549f9798674301c5af2c91581194091/tenor.gif')
+  const [firstName, setFirstName] = React.useState('Loading...')
+  const [lastName, setLastName] = React.useState('Loading...')
+  const [email, setEmail] = React.useState('Loading...')
+  
+  React.useEffect(() => {
+    fetch(`https://reqres.in/api/users/${props.userId}`)
+    .then((res) => res.json())
+    .then((json) => {
+      setAvatar(json.data.avatar)
+      setFirstName(json.data.first_name)
+      setLastName(json.data.last_name)
+      setEmail(json.data.email)
+    })
+    .catch((err) => console.log(err))
+  })
```

Comment:
The `useEffect` hook takes in two arguments - an effect function to fire, and an array of dependencies. If nothing is passed, the function will fire each time the component renders. In this case, if this component ever re-rendered, we'd do some un-needed fetching. In this case, we want to re-fetch anytime the userId changes, so you'd want to include the `userId` prop as a dependency.

Reply:
```suggestion
  }, [props.userId])
```